src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/TransportCreation/TcpTransportCreationHandler.cs

```diff
@@ -13,93 +12,51 @@
+    /// <summary>
+    /// Handler for connecting via TCP.
+    /// </summary>
+    internal sealed class TcpTransportCreationHandler : IReturningHandler<ConnectionHandlerContext, Stream>
     {
         private const int KeepAliveIntervalSeconds = 1;
         private const int KeepAliveTimeSeconds = 30;
 
         #if NET8_0_OR_GREATER
         private static readonly TimeSpan DefaultPollTimeout = TimeSpan.FromSeconds(30);
         #else
-        private const int DefaultPollTimeout = 30 * 100000; // 30 seconds as microseconds
+        private const int DefaultPollTimeout = 30 * 1_000_000; // 30 seconds as microseconds
         #endif
 
         /// <inheritdoc />
-        public IHandler<ConnectionHandlerContext> NextHandler { get; set; }
-
-        /// <inheritdoc />
-        public async ValueTask Handle(ConnectionHandlerContext context, bool isAsync, CancellationToken ct)
+        public async ValueTask<Stream> Handle(ConnectionHandlerContext parameters, bool isAsync, CancellationToken ct)
         {
-            Debug.Assert(context.DataSource is not null, "context.DataSource is null");
-
-            try
-            {
-                // @TODO: Build CoR for handling the different protocols in order
-                if (context.DataSource.ResolvedProtocol is DataSource.Protocol.TCP)
-                {
-                    context.ConnectionStream = await HandleTcpRequest(context, isAsync, ct).ConfigureAwait(false);
-                }
-                else
-                {
-                    throw new NotImplementedException();
-                }
-            }
-            catch (Exception e)
-            {
-                context.Error = e;
-                return;
-            }
-
-            // Every physical connection has a Unique ID which is used by the rest 
-            // of the connection for tracing on the server side, to correlate with 
-            // the connectivity telemetry and diagnosis.
-            context.ConnectionId = Guid.NewGuid();
-
-            if (NextHandler is not null)
+            // This handler cannot process if the protocol does not contain TCP
+            if (parameters.DataSource.ResolvedProtocol is not (DataSource.Protocol.Admin or DataSource.Protocol.TCP or DataSource.Protocol.None))
             {
-                await NextHandler.Handle(context, isAsync, ct).ConfigureAwait(false);
+                return null;
             }
-        }
 
-        private ValueTask<Stream> HandleNamedPipeRequest()
-        {
-            throw new NotImplementedException();
-        }
-
-        private ValueTask<Stream> HandleSharedMemoryRequest()
-        {
-            throw new NotImplementedException();
-        }
-
-        private async ValueTask<Stream> HandleTcpRequest(ConnectionHandlerContext context, bool isAsync, CancellationToken ct)
-        {
             ct.ThrowIfCancellationRequested();
 
             // DNS lookup
             IPAddress[] ipAddresses = isAsync
```

Comment:
Is this the only one time Dns lookup will happen for connection attempt(s)? If yes, no change is needed.
If no, you may consider this Dns Lookup result be stored somewhere. 

One of the important things to consider is that Dns calls often timeout, i.e. they take 5 secs+ to return, I would recommend this result be cached in context by Data Source parser for the connection event if this will lead to timeout multiple times for the same connection open request.

Reply:
Yeah, this is something that will likely be addressed in a later PR.